context/AuthContextProvider.tsx

```diff
@@ -50,8 +50,12 @@ export const AuthContextProvider = ({
       getUser();
       return;
     }
+
     setIsSignedIn(true);
-  }, [user]);
+    if (pathname.startsWith('/admin')) {
+      !user.labels.includes('admin') && router.push('/');
+    }
+  }, [user, pathname]);
 
```

Comment:
Just some ideas to think about: Should we write out other edge cases such as 1) if user.labels is undefined or not yet populated
2) if the logic has multiple stated updates (i.e. user or pathname changing frequently causing multiple redirects (to prevent adding to the history stack as a user can go forward and backward)

Reply:
1.  The way the useEffect is written user will be populated if it hits this point

```
    if (user.id === '' || user.email === '') {
      getUser();
      return;
    }
    ```

2. This is a question I'd like to defer to @shashilo ... right now this is the only way I was able to get it to work due when we change pathname it loads the data again (using cached data) but we still need to wait for it to load.

Reply:
@vmaineng brings up a great point. If this is a case in our app, we should write it out by making a unit test for it.

Reply:
✅ pathname helper function added to utils with a unit test to test pathname changing.